snakemake/common.py

```diff
@@ -54,6 +55,43 @@ def get_uuid(name):
     return uuid.uuid5(UUID_NAMESPACE, name)
 
 
+def get_file_hash(filename, algorithm="sha256"):
+    """find the SHA256 hash string of a file. We use this so that the
+       user can choose to cache working directories in storage.
+    """
+    from snakemake.logging import logger
+
+    if os.path.exists(filename):
```

Comment:
The if is superfluous. `open()` already throws a proper exception.

Reply:
Fixed.